build-tooling.mk

```diff
@@ -144,37 +154,43 @@ publish:
 lint:
 ifneq ($(strip $(COMPONENT)),.)
 	cp .golangci.yaml $(COMPONENT)
-	$(DOCKER) build . -f Dockerfile --target lint --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
+	$(DOCKER) build . -f $(COMPONENT_DOCKERFILE) --target lint --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
 	rm -rf $(COMPONENT)/.golangci.yaml
 else
-	$(DOCKER) build . -f Dockerfile --target lint --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
+	$(DOCKER) build . -f $(COMPONENT_DOCKERFILE) --target lint --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
 endif
 
 .PHONY: fmt
 # Run go fmt against code
 fmt:
-	$(DOCKER) build . -f Dockerfile --target fmt --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
+	$(DOCKER) build . -f $(COMPONENT_DOCKERFILE) --target fmt --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
 
 .PHONY: vet
 # Perform static analysis of code
 vet:
-	$(DOCKER) build . -f Dockerfile --target vet --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
+	$(DOCKER) build . -f $(COMPONENT_DOCKERFILE) --target vet --build-arg COMPONENT=$(COMPONENT) --build-arg GOPROXY_ARG=$(GOPROXY)
 
 .PHONY: test
 # Run tests
 test: fmt vet
```

Comment:
previously if I wanted to run individual make targets like for example `test`, I could easily do that by running `make test COMPONENT=/path/to/foo`, but now I also need to pass the path to component dockerfile by setting `COMPONENT_DOCKERFILE` var which is computed somewhere in the top level make target, can we provide a similar experience as before to run individual make targets?

Reply:
Added changes to support running targets individually. IMO, this is a bit complicated scenario. 
Due to multiple permutations and nested order in which last mile targets like "fmt" and "vet" could be run:

- `make fmt`
- `make test` -> Triggers fmt
- `make docker-build-all` -> Triggers build -> test -> fmt

It's difficult to maintain and pass a state flag like `KEEP_COMPONENT_DOCKERFILE`, which should be conditionally set based on which nested level the target was triggered at. Eg, on triggering `make test` directly, fmt target should not perform dockerfile cleanup, even if `KEEP_COMPONENT_DOCKERFILE` is set to false globally. To determine whether current target is the final state / its yet another level in the nested invocation (docker-build-all) is not feasible using a single state variable. Thus, introduced another flag `SKIP_INTERMEDIATE_CLEANUP` to handle this.

Reply:
Another option as suggested by @sathyanarays is to make the `cleanup-component-dockerfile` target independent of other target invocations, and we expect the users to explicitly trigger it. I think this is a valid solution to keep things simple.

Reply:
Added these changes into build-tooling repo branch for future reference: https://github.com/vmware-tanzu/build-tooling-for-integrations/commit/827a5ae9e162201e422ca603cc6565f2e41a25d7